### PR TITLE
Molecular networking 

### PR DESCRIPTION
**Pull request by robinschmid**
Optimized the parameters and algorithms

### CODE REVIEW

**Comment by robinschmid** [comment]
moved molecular networking package outside

File: src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SignalFiltersParameters.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+/**
+ * Filters to apply to spectra
+ *
+ * @author Robin Schmid <a href="https://github.com/robinschmid">https://github.com/robinschmid</a>
+ */
+public class SignalFiltersParameters extends SimpleParameterSet {
+
+  public static final OptionalParameter<DoubleParameter> removePrecursor = new OptionalParameter<>(
+      new DoubleParameter("Remove residual precursor m/z",
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SignalFiltersParameters.java:44]: Parameter name + value don't really add up. Sounds like you are removing m/z 10 from the spectra.

Maybe call it "Remove singals around precursor (width)"

File: src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SpectralNetworkingParameters.java [hunk 2]
```diff
@@ -53,38 +54,34 @@ public class MS2SimilarityParameters extends SimpleParameterSet {
...
       "Check MS2 neutral loss similarity",
       "Generates a list of m/z differences and calculates cosine similarity",
       new NeutralLossSimilarityParameters(), false);
 
-  public MS2SimilarityParameters() {
-    super(
-        new Parameter[]{FEATURE_LISTS, MZ_TOLERANCE, ONLY_BEST_MS2_SCAN, MIN_HEIGHT, MIN_DP,
-            MIN_MATCH, MIN_COSINE_SIMILARITY, MODIFICATION_AWARE_COSINE,
-            CHECK_NEUTRAL_LOSS_SIMILARITY});
+  public static final OptionalParameter<DoubleParameter> MAX_MZ_DELTA = new OptionalParameter<>(
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SpectralNetworkingParameters.java:72]: couldnt this also go in the SignalFiltersParameters?

File: src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SpectralNetworkingTask.java [hunk 3]
```diff
@@ -348,24 +360,32 @@ public static int calcOverlap(List<DataPoint[]> aligned) {
...
       if (onlyBestMS2Scan) {
-        checkRowsBest(mapCosineSim, mapNeutralLoss, rows);
+        checkRowsBestMs2(mapCosineSim, mapNeutralLoss, rows);
       } else {
         checkAllFeatures(mapCosineSim, mapNeutralLoss, rows);
       }
-      LOG.info(MessageFormat.format(
+      logger.info(MessageFormat.format(
           "MS2 similarity check on rows done. MS2 cosine similarity={0}, MS2 neutral loss={1}",
           mapCosineSim.size(), mapNeutralLoss.size()));
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SpectralNetworkingTask.java:373]: `mapCosineSim.size()?` dont you want to print the actual similarity here?

**Review by SteffenHeu — COMMENTED** [review]
some comments

File: src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SignalFiltersParameters.java (hunk 1 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SignalFiltersParameters.java:44]: its actually a delta mz

File: src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SpectralNetworkingTask.java (hunk 3 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/dataprocessing/group_spectral_networking/SpectralNetworkingTask.java:378]: This is the total number of edges. Added this to the message

**Pull request merged**